zerver/lib/push_notifications.py

```diff
@@ -744,6 +778,38 @@ def initialize_push_notifications() -> None:
             "See https://zulip.readthedocs.io/en/latest/"
             "production/mobile-push-notifications.html"
         )
+        return
+
+    if uses_notification_bouncer():  # nocoverage
+        # If we're using the notification bouncer, check if we can
+        # actually send push notifications.
+
+        try:
+            realms = send_realms_only_to_push_bouncer()
+        except Exception:
+            # TODO: Catch specific exceptions.
+            for realm in Realm.objects.filter(push_notifications_enabled=True):
+                do_set_realm_property(realm, "push_notifications_enabled", False, acting_user=None)
+            return
```

Comment:
The idea is that if the self-hosted server gets a fatal, non-transient error trying to reach the push notifications service, it is very possible that you've got a firewall in the way or you accidentally corrupted your credentials in your config file or whatever, and indeed it's not working and will continue to not be working.

Hence the TODO -- what type of exception is thrown probably impacts. But I don't think it's a wrong first pass to just mark it as "believed not working" here if we can't reach the bouncer.

Reply:
So maybe let's just replace the comment with something like this:
```
# An exception was thrown trying to ask the bouncer service whether we can send push notifications or not. 
# There may be certain transient failures that we could ignore here, but the default explanation is that there 
# is something wrong either with our credentials being corrupted or our ability to reach the bouncer service 
# over the network, so we immediately move to reporting push notifications as likely working, as whatever failed
# here is likely to also fail when trying to send a push notification.

```

Oh and also we should probably do `logging.exception` or something so that appropriate logs are generated of the specific failure.